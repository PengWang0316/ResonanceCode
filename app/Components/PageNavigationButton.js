import React from "react";
import PropTypes from "prop-types";
import { Link } from "react-router-dom";

const PageNavigationButton = (props) => {
  if(props.isEmptyContent) return null;
  else {
    return (
      <div className="pageBtnDiv text-right">
        {props.startNumber>1 && <Link className="btn btn-info loginButton pageBtn" to={{
          pathname: "/reading",
          search: `?start=${props.startNumber*1-5>1?props.startNumber*1-5:1}`
        }} ><i className="fa fa-backward" />Previous</Link>}
        <Link className="btn btn-info loginButton pageBtn" to={{
          pathname: "/reading",
          search: `?start=${props.startNumber*1+5}`
        }}>Next<i className="fa fa-forward" /></Link>
      </div>
    );
  }
};
PageNavigationButton.propTypes = {
  startNumber: PropTypes.string.isRequired,
  isEmptyContent: PropTypes.bool.isRequired
};
export default PageNavigationButton;
