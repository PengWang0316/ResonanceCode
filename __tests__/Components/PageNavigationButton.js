import React from 'react';
import renderer from 'react-test-renderer';
import { shallow } from 'enzyme';

import PageNavigationButton from '../../app/Components/PageNavigationButton';

jest.mock('react-router-dom', () => ({ Link: 'Link' }));

describe('PageNavigationButton test', () => {
  const defaultProps = {
    startNumber: '2',
    isEmptyContent: false
  };
  const getShallowComponent = (props = defaultProps) =>
    shallow(<PageNavigationButton {...props} />);

  test('PageNavigationButton snapshot', () => expect(renderer.create(<PageNavigationButton {...defaultProps} />).toJSON()).toMatchSnapshot());

  test('PageNavigationButton empty content snapshot', () => expect(renderer.create(<PageNavigationButton startNumber="2" isEmptyContent />).toJSON()).toMatchSnapshot());

  test('Link search property test', () => {
    const component = getShallowComponent();
    expect(component.find('Link').at(0).prop('to').search).toBe('?start=1');
    const newComponent = getShallowComponent({ isEmptyContent: false, startNumber: '7' });
    expect(newComponent.find('Link').at(0).prop('to').search).toBe('?start=2');
  });
});
