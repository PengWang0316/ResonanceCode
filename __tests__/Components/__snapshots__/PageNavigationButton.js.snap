// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PageNavigationButton test PageNavigationButton empty content snapshot 1`] = `null`;

exports[`PageNavigationButton test PageNavigationButton snapshot 1`] = `
<div
  className="pageBtnDiv text-right"
>
  <Link
    className="btn btn-info loginButton pageBtn"
    to={
      Object {
        "pathname": "/reading",
        "search": "?start=1",
      }
    }
  >
    <i
      className="fa fa-backward"
    />
    Previous
  </Link>
  <Link
    className="btn btn-info loginButton pageBtn"
    to={
      Object {
        "pathname": "/reading",
        "search": "?start=7",
      }
    }
  >
    Next
    <i
      className="fa fa-forward"
    />
  </Link>
</div>
`;
